docs/source/testing/testing.md

```diff
@@ -43,27 +72,76 @@ it('fetches single launch', async () => {
+For examples of both integration and end-to-end testing we recommend checking out the [tests included in the Apollo fullstack tutorial](https://github.com/apollographql/fullstack-tutorial/tree/master/final/server/src/__tests__).
+## End-to-end testing
 
-In addition to `query`, the first argument to `executeOperation` can take `operationName`, `variables`, `extensions`, and `http` keys.
 
-Note that errors in parsing, validating, and executing your operation are returned in the `errors` field of the result (just like in a GraphQL response) rather than thrown.
+Instead of bypassing the HTTP layer, you might want to fully run your server and test it with a real HTTP client. Apollo Server doesn't have any built-in support for this at this time.
 
-You don't have to call [`start`](../../api/apollo-server/#start) before calling `executeOperation`; it will be called automatically for you if it has not been called, and any startup errors will be thrown.
+You can run operations against your server using a combination of any HTTP or GraphQL client such as [`supertest`](https://www.npmjs.com/package/supertest) or [Apollo Client's HTTP Link](https://www.apollographql.com/docs/react/api/link/apollo-link-http/) . There are also community packages available such as [`apollo-server-integration-testing`](https://www.npmjs.com/package/apollo-server-integration-testing), which uses mocked Express request and response objects.
 
+Below is an example of writing an end-to-end test using the [`apollo-server-express` package](../integrations/middleware/#apollo-server-express) and `supertest`:
 
-## End-to-end testing
+```js:title=e2e.test.js
+// we import our server instance
+const { server } = require('./server.js');
+const express = require('express');
+const { createServer } = require('http');
+
+// we will use supertest to test our server
+const request = require('supertest');
 
-Instead of bypassing the HTTP layer, you may just want to fully run your server and test it with a real HTTP client.
+// this is the query we use for our test
+const queryData = {
+  query: `query sayHello($name: String) {
+    hello(name: $name)
+  }`,
+  variables: { name: 'world' },
+};
 
-Apollo Server doesn't have any built-in support for this. You can combine any HTTP or GraphQL client such as [`supertest`](https://www.npmjs.com/package/supertest) or [Apollo Client's HTTP Link](https://www.apollographql.com/docs/react/api/link/apollo-link-http/) to run operations against your server. There are also community packages available such as [`apollo-server-integration-testing`](https://www.npmjs.com/package/apollo-server-integration-testing) which uses mocked Express request and response objects.
+// create a new http server for testing
+const createTestServer = async () => {
```

Comment:
maybe createTestHttpServer, to be clear that this is not the same as the ApolloServer?

I do also feel like you probably will want to create and `.start()` the ApolloServer from the beforeAll rather than in the other file.